### DISCUSSION TITLE
Still no semi colon in JS for v5 ?

### DISCUSSION THREAD

**Discussion by MathRobin** [twbs/bootstrap#31321] [General] [answered]
Hi,

You still keep no-semi colon style for v5 ? Why ? Most people (majority ?) still consider that semicolon are needed and are not an option. And they are not, I don't need to remember you that there is still ASI. So why not use them ?
Innocent question, not a troll.

**Comment by Luke-zhang-04** [comment]
The code compiles to distribution with semicolons, so it doesn't change the way Bootstrap works in production.

**Reply by MathRobin** [reply]
Because there is something which can manage sh*t, is not an excuse to make sh*t. It's like if you say, I can make something stupid with matches, there are firemen next door.

**Comment by mdo** [comment, marked as answer]
It's still our style, yes. No plans to change this.

**Reply by MathRobin** [reply]
OK, seems legit. Even if I still don't understand why bootstrap is the only project with this style...

**Comment by Johann-S** [comment]
it seems we are not alone according to: https://standardjs.com/ @MathRobin

**Reply by MathRobin** [reply]
It seems that you're true. Still understand why the f*** you do it like this. What are the pros to not use the semi-colons ? "it's our style" is not really a convincing argument

**Reply by Johann-S** [reply]
You can read more about it here:
https://blog.izs.me/2010/12/an-open-letter-to-javascript-leaders-regarding
http://inimino.org/~inimino/blog/javascript_semicolons

**Reply by MathRobin** [reply]
The blog article from Issac is really clear. Thanks. It looks like "eyh look I'm a badass, I know what I do and I have to show I'm superior to you." (mostly the part when he said he pissed on beginners). But other arguments are good too. I can understand this choice

**Reply by Johann-S** [reply]
Haha I'm not the author :)
We kept this rule because it's the legacy of the previous Bootstrap versions made by @fat and the others

**Reply by MathRobin** [reply]
Yes I know you're not Isaacs. I end here, don't understand but maybe you are not wrong. Maybe.

**Reply by Deckluhm** [reply]
The cognitive load added by semicolon (and more generally all optional characters) is the number one reason I don't use semicolon in JS (and most optional characters in Ruby as well).

You obviously need to get used to it but once you are, it makes code simpler and easier to read.

You can argue there are some edge cases and you are right but I feel like they are too rare to keep semicolon everywhere (and tools like ESLint/Prettier/TypeScript will probably warn you about edge cases).

**Reply by MathRobin** [reply]
The argument of the cognitive load is interesting. I read it like it's "I have to think about washing my hands after eating. What a cognitive load". Hell, it's so obvious that it's no more a cognitive load...

The edge cases you told are not a good argument too in favor of using semi colon. Because for the most they are rare and like you said, there are linters, transpilers, etc.

**Reply by mohe2015** [reply]
https://en.wikipedia.org/wiki/JavaScript_syntax#Whitespace_and_semicolons might be interesting to read. I basically agree with MathRobin now that I read it but I also agree with the others. So I assume it's mostly personal taste and big projects have to choose one which is no semicolons in this case. Also it seems like tooling prevents most real issues.

**Reply by Deckluhm** [reply]
> The argument of the cognitive load is interesting. I read it like it's "I have to think about washing my hands after eating. What a cognitive load". Hell, it's so obvious that it's no more a cognitive load...

It actually became obvious because you learned to do it, but it wasn't obvious in the first place (I'm learning code to a beginner and it's definitely not always obvious to him when to put or not to put comas and semicolons).

Even if I have a personal preference for "no semicolon", I think there are pros and cons in using or not using them.

I'm not trying to convince you not to use semicolon, I just want you to understand there are valid reasons not to use them (maybe not valid enough for you :wink:).

**Reply by MathRobin** [reply]
There are probably some good reasons, I'm just still searching them.
Maybe I guess I think it's obvious to use them, because in many/most(/all ?) human languages, there is always a dot at the end of a sentence which indicates the end of the sentence, like the semi colons indicate the end of a line. So maybe it influences me...
